test/dotnet-compile.Tests/CompilerTests.cs

```diff
@@ -167,6 +167,24 @@ public void DepsDependencyContextIsValidOnBuild()
                 runCommand.Execute().Should().Pass();
         }
 
+        [Fact]
+        public void CanSetOutputAssemblyName()
+        {
+            var testInstance =
+                TestAssetsManager
+                    .CreateTestInstance("LibraryWithOutputAssemblyName")
+                    .WithLockFiles();
+
+            var root = testInstance.TestRoot;
+            var outputDir = Path.Combine(root, "bin");
+            var testProject = ProjectUtils.GetProjectJson(root, "LibraryWithOutputAssemblyName");
+            var buildCommand = new BuildCommand(testProject, output: outputDir, framework: DefaultFramework);
+            var result = buildCommand.ExecuteWithCapturedOutput();
+            result.Should().Pass();
+
+            new DirectoryInfo(outputDir).Should().HaveFiles(new [] { "MyLibrary.dll" });
```

Comment:
Actually I also see this in the deps file: 
  "libraries": {
    "LibraryWithOutputAssemblyName/1.0.0": {
      "type": "project",
      "serviceable": false,
      "sha512": ""
    },

and also the same dll on the disk so am not sure if this is right...